packages/cli/src/workflows/workflows.services.ee.ts

```diff
@@ -0,0 +1,73 @@
+import { DeleteResult, EntityManager, In, Not } from 'typeorm';
+import { Db } from '..';
+import { SharedWorkflow } from '../databases/entities/SharedWorkflow';
+import { User } from '../databases/entities/User';
+import { WorkflowEntity } from '../databases/entities/WorkflowEntity';
+import { RoleService } from '../role/role.service';
+import { UserService } from '../user/user.service';
+import { WorkflowsService } from './workflows.services';
+
+export class EEWorkflowsService extends WorkflowsService {
+	static async isOwned(
+		user: User,
+		workflowId: string,
+	): Promise<{ ownsWorkflow: boolean; workflow?: WorkflowEntity }> {
+		const sharing = await this.getSharing(user, workflowId, ['workflow', 'role'], {
+			allowGlobalOwner: false,
+		});
+
+		if (!sharing || sharing.role.name !== 'owner') return { ownsWorkflow: false };
+
+		const { workflow } = sharing;
+
+		return { ownsWorkflow: true, workflow };
+	}
+
+	static async getSharings(
+		transaction: EntityManager,
+		workflowId: string,
+	): Promise<SharedWorkflow[]> {
+		const workflow = await transaction.findOne(WorkflowEntity, workflowId, {
+			relations: ['shared'],
+		});
+		return workflow?.shared ?? [];
+	}
+
+	static async pruneSharings(
+		transaction: EntityManager,
+		workflowId: string,
+		userIds: string[],
+	): Promise<DeleteResult> {
+		return transaction.delete(SharedWorkflow, {
+			workflow: { id: workflowId },
+			user: { id: Not(In(userIds)) },
+		});
+	}
+
+	static async share(
+		transaction: EntityManager,
+		workflow: WorkflowEntity,
+		shareWithIds: string[],
+	): Promise<SharedWorkflow[]> {
+		const [users, role] = await Promise.all([
+			UserService.getByIds(transaction, shareWithIds),
+			RoleService.trxGet(transaction, { scope: 'workflow', name: 'user' }),
+		]);
+
+		const newSharedWorkflows = users.reduce((acc, user) => {
+			if (user.isPending) {
+				return acc;
+			}
+			acc.push(
+				Db.collections.SharedWorkflow.create({
+					workflow,
+					user,
+					role,
+				}),
+			);
+			return acc;
+		}, [] as SharedWorkflow[]);
```

Comment:
Nit: The type for the accumulator is provided [as a generic](https://github.com/microsoft/TypeScript/blob/afee8bf9682dfb3ed927fdc7756d96a19851b811/lib/lib.es5.d.ts#L1386) to `reduce`